src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioUploadServiceImpl.java

```diff
@@ -175,24 +158,26 @@ private void versionUpload(List<MediaEntity> mediaToReplace,
             newMediaEntity,
             objectRecordStatusEntity
         );
-        for (MediaEntity mediaEntity : mediaToReplace) {
+        for (MediaEntity mediaEntity : mediaToSupersede) {
             deleteMediaLinking(mediaEntity);
         }
 
         logApi.audioUploaded(addAudioMetadataRequest);
     }
 
-    private List<MediaEntity> filterAudioFilesWithDifferentFileSizes(Collection<MediaEntity> mediaEntities, long expectedFileSize) {
-        List<MediaEntity> mediaEntitiesToVersion = new ArrayList<>();
-        for (MediaEntity mediaEntity : mediaEntities) {
-
-            // if file size is not expected
-            if (expectedFileSize != mediaEntity.getFileSize()) {
-                mediaEntitiesToVersion.add(mediaEntity);
-            }
+    private String computeChecksum(MultipartFile audioFileStream) {
+        try (var digestInputStream = new DigestInputStream(new BufferedInputStream(audioFileStream.getInputStream()), audioDigest.getMessageDigest())) {
+            ByteStreams.exhaust(digestInputStream); // We must consume the stream before calculating the checksum
```

Comment:
Do we need to do this?  Doesnt the DigestInputStream allow us to get a checksum from the stream?

Reply:
The checksum is computed incrementally as the stream is consumed. So to compute the checksum for the entire data, we first need to consume the entire stream.